### PR TITLE
Create Contact Me Section and Fetch JSON Messages

### PR DESCRIPTION
**Pull request by areetaw**
With week 3's server-side development, there is now a `contact` section with the ability to fetch JSON messages. 

In order to fetch the JSON message, the `DataServlet` creates the `ArrayList` and returns it as a JSON string. Then, the `script.js` uses `fetch` to grab the string from the server. All strings are all hard-coded.

image

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -15,18 +15,41 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import com.google.gson.Gson;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
```
Comment (mrboxtobox) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:24]: Could you order the imports?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -15,18 +15,41 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Areeta!</h1>");
+
+    // Convert the messages to JSON
+    messages = new ArrayList<>();
+    messages.add("boop bop beep");
+    messages.add("bopity beep boop");
+    messages.add("beep boop bop");
+    String json = convertToJsonUsingGson(messages);
```
Comment (mrboxtobox) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: LOL

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -15,18 +15,41 @@
...
+
+    // Send the JSON as the response
```
Comment (mrboxtobox) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: It's usually recommended to end comments with a period.

File: portfolio/src/main/webapp/images/svg/icons8-edit.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...9999999H18.88z" transform="rotate(45 22.068 33.441)"/><path fill="#8d6c9f" d="M60.59,15.9a4,4,0,0,0-1.17-2.83L50.93,4.59a4.09,4.09,0,0,0-5.66,0L41,8.83a2,2,0,0,0-2.83,0l-5.66,5.66a2,2,0,0,0,0,2.83l.05.05L5.52,44.4a3,3,0,0,0-.87,1.88L4,54.68l-.7,4.92a1,1,0,0,0,1,1.14H4.4L9.32,60l8.45-.62a3,3,0,0,0,1.9-.87l27.08-27a2,2,0,0,0,2.77,0l5.66-5.66a2,2,0,0,0,0-2.83l4.24-4.24A4,4,0,0,0,60.59,15.9Zm-48.69,32,.58,2.88a1,1,0,0,0,.78.78l2.88.58,1.07,5.34L9.59,58,6,54.41l.61-7.6ZM19,56.35l-1-4.77L34.67,35a1,1,0,0,0-1.41-1.41L16.66,50.17,14.3,49.7l-.47-2.36L27.59,33.58a1,1,0,0,0-1.41-1.41L12.42,45.93,7.75,45,34,18.78,45.32,30.09ZM52.34,25.8h0a1,1,0,0,0-1.41,0l-1.41,1.41a1,1,0,0,0,0,1.41h0L48.1,30h0L34,15.9l1.41-1.41a1,1,0,0,0,1.41,0l1.41-1.41a1,1,0,0,0,0-1.41l1.41-1.41.71.71L53.05,23.68l.71.71ZM58,17.31l-4.24,4.24L42.44,10.24,46.69,6a2,2,0,0,1,2.83,0L58,14.49a2,2,0,0,1,0,2.83Z"/><path fill="#8d6c9f" d="M40.32 15.19l-1.41 1.41A1 1 0 1 0 40.32 18l1.41-1.41a1 1 0 0 0-1.41-1.41zM43.86 18.73l-1.41 1.41a1 1 0 1 0 1.41 1.41l1.41-1.41a1 1 0 0 0-1.41-1.41zM47.39 22.26L46 23.68a1 1 0 1 0 1.41 1.41l1.41-1.41a1 1 0 0 0-1.41-1.41zM31.84 26.51L29 29.33a1 1 0 1 0 1.41 1.41l2.83-2.83a1 1 0 0 0-1.41-1.41z"/></svg>
```
Comment (mrboxtobox) [inline portfolio/src/main/webapp/images/svg/icons8-edit.svg:1]: Nice! Good to se you playing around with SVGs =)

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -36,3 +38,12 @@ function handleOnScroll() {
...
 }
+
+/**
+ * Adds messages.
+ */
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((message) => {
+    document.getElementById('message-container').innerText = message;
+  });
+}
```
Comment (mrboxtobox) [inline portfolio/src/main/webapp/script.js:49]: You can end the file with a newline here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (areetaw) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:24]: Yes, thanks!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (areetaw) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: :)))

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (areetaw) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: Gotcha, thank you!

File: portfolio/src/main/webapp/images/svg/icons8-edit.svg (hunk 4 above)
Comment (areetaw) [reply portfolio/src/main/webapp/images/svg/icons8-edit.svg:1]: Haha, they're looking pretty good! :)

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (areetaw) [reply portfolio/src/main/webapp/script.js:49]: Fixed!

**Pull request merged**